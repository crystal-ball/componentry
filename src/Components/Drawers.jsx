import React from 'react';
import { Drawer } from '../../lib';

export default function Drawers() {
  return (
    <div>
      <div className="row">
        <div className="col-12">
          <h2>Drawers</h2>
        </div>
        <div className="col-6">
          <h4>Subcomponents</h4>
          <Drawer>
            <Drawer.Trigger>
              <span className="chevron-icon" /> Toggle Drawer
            </Drawer.Trigger>
            <Drawer.Content>
              The Motion Picture Academy refused to nominate Tron (1982) for a
              special-effects award because, according to director Steven Lisberger,
              “The Academy thought we cheated by using computers”
            </Drawer.Content>
          </Drawer>
        </div>
        <div className="col-6">
          <h4>Parameter Values</h4>
          <Drawer Trigger="Toggle Drawer" Content="Short hand syntax" />
        </div>
      </div>
    </div>
  );
}
