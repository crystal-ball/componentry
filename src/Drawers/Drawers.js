// @flow
import withActive from '../withActive'
import withTheme from '../withTheme'
import activeContainer from '../active-container-factory'
import activeContent from '../active-content-factory'
import activeTrigger from '../active-trigger-factory'

const makeDrawer = name => {
  const elem = name.slice(0, 1).toLocaleLowerCase() + name.slice(1)

  const Content = activeContent({
    element: elem,
    arias: { id: true, hidden: true },
  })
  const withActiveContent = withActive(withTheme(`${name}Content`, Content))

  const Trigger = activeTrigger({
    element: elem,
    arias: { controls: true, expanded: true },
  })
  const withActiveTrigger = withActive(withTheme(`${name}Trigger`, Trigger))

  const Element = withTheme(
    name,
    activeContainer({
      element: elem,
      name,
      Content: withActiveContent,
      Trigger: withActiveTrigger,
    }),
  )

  Element.Content = withActiveContent
  Element.Trigger = withActiveTrigger

  return Element
}

const Drawer = makeDrawer('Drawer')
const Accordion = makeDrawer('Accordion')

export { Accordion, Drawer }
