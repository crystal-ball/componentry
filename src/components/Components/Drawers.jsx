// @flow
import React from 'react'

import { Drawer } from '../../../lib'

export default () => (
  <div className="mb-5">
    <div className="row">
      <div className="col-12">
        <p className="lead">Drawer component...</p>
        <Drawer>
          <Drawer.Trigger>
            <svg role="img" className="icon chevron font">
              <use href="#chevron" />
            </svg>
            Toggle Drawer
          </Drawer.Trigger>
          <Drawer.Content>
            The Motion Picture Academy refused to nominate Tron (1982) for a
            special-effects award because, according to director Steven Lisberger,
            “The Academy thought we cheated by using computers”
          </Drawer.Content>
        </Drawer>
      </div>
    </div>
  </div>
)
