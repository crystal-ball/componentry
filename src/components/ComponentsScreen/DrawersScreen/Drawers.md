<p className="lead">Drawer component...</p>
<Drawer>
  <Drawer.Trigger>
    <Icon id="chevron" />
    Toggle Drawer
  </Drawer.Trigger>
  <Drawer.Content>
    The Motion Picture Academy refused to nominate Tron (1982) for a
    special-effects award because, according to director Steven Lisberger,
    “The Academy thought we cheated by using computers”
  </Drawer.Content>
</Drawer>
<h2>Accordion</h2>
<Accordion>
  <Accordion.Trigger activeId="one">Trigger One</Accordion.Trigger>
  <Accordion.Content activeId="one">Content One</Accordion.Content>
  <Accordion.Trigger activeId="two">Trigger Two</Accordion.Trigger>
  <Accordion.Content activeId="two">Content Two</Accordion.Content>
</Accordion>
