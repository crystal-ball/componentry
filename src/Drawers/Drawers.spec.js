import React from 'react'
import { render } from 'enzyme'

import { Drawer } from './Drawers'
import activationTestSuite from '../utils-test/activation-tests'
import elementTests from '../utils-test/element-tests'

describe('<Drawer />', () => {
  activationTestSuite(Drawer)
  elementTests(Drawer)
  elementTests(Drawer.Trigger)
  elementTests(Drawer.Content)
})

// Snapshots
// ---------------------------------------------------------------------------
describe('<Drawer /> snapshots', () => {
  it('renders correctly', () => {
    const tree = render(
      <Drawer>
        <Drawer.Trigger>Trigger</Drawer.Trigger>
        <Drawer.Content>Content</Drawer.Content>
      </Drawer>,
    )

    expect(tree).toMatchSnapshot()
  })
})
