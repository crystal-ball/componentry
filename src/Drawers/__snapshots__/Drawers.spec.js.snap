// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> snapshots renders correctly 1`] = `
<div
  class="drawer"
  data-id="guid"
>
  <button
    aria-controls="guid"
    aria-expanded="false"
    class="drawer-toggle btn-anchor"
    type="button"
  >
    Trigger
  </button>
  <div
    aria-hidden="true"
    class="drawer-content"
    id="guid"
  >
    Content
  </div>
</div>
`;
